sklearn/tests/test_isotonic.py

```diff
@@ -535,3 +536,19 @@ def test_isotonic_thresholds(increasing):
         assert all(np.diff(y_thresholds) >= 0)
     else:
         assert all(np.diff(y_thresholds) <= 0)
+
+
+def test_infinite_probabilities():
```

Comment:
Currently this test passes on master.

Reply:
> Is this happening because there are some elements in np.sum(proba, axis=1) that is zero? (In predict_proba)

That might be possible. I am trying to figure out the values with which the issue can be recreated but it only shows up with a large dataset.
What alternatives can we take?

Reply:
See if `np.sum(proba, axis=1)` is zero for the large dataset and see why this is happening.

Reply:
okay

Reply:
Hi @Thomas9292, so it seems like the infinity is returned by the classifiers 
https://github.com/scikit-learn/scikit-learn/blob/f642ff79fd2d8bf3c41e54a03ee9aeec502732da/sklearn/calibration.py#L345-L347

One way to show this issue is by serializing. Would that be a good idea to solve this issue?

Reply:
@Thomas9292 A bit more context: we noticed that the infinities were propagating up from CalibratedClassifier instances.

If the goal is just to show (without committing a big data set to the repo) that `predict_proba` can return infinite probabilities, it will be sufficient to serialize the `CalibratedClassifier` objects (or the `CalibratedClassifierCV` itself) trained on the large data sets. The serialized objects can just be read in a test to reproduce the infinite probability bug.

If the goal is to nail down exactly where this problem first arises, then the alternative will be a fair amount of archaeology--digging through the implementation to discern the ultimate source of these infinities.

What makes more sense to you?

Reply:
Are you trying to ping me?

I am trying to guide us to: "How do we generate a dataset so we can have a non-regression test?" We can not use a serialized model, because we do not support unserializing between different versions of sklearn. Understanding the underlying issue, can help us generate this dataset.

After going through the data, it looks like `np.inf` comes from `IsotonicRegression`:

```py
from sklearn.isotonic import IsotonicRegression
import numpy as np

X = np.array([0., 4.1e-320, 4.4e-314, 1.])
y = np.array([0.42, 0.42, 0.44, 0.44])

iso = IsotonicRegression().fit(X, y)

iso.predict(np.array(iso.predict(np.array([0, 2.1e-319, 5.4e-316, 1e-10]))))
# array([0.42,  inf,  inf, 0.44])
```

In this case the `np.clip` from this PR would map the middle two values to 1, which which would be incorrect.

Reply:
Thanks for clarifying that @Thomas9292. I am working on the categorical feature support issue currently. I'll try to resolve this later. Thanks for the reviews!

Reply:
My username is @thomasjpfan 😅

Reply:
Oops sorry about that :(

Reply:
No problem @yashika51, always happy to help and even happier to get credit for @thomasjpfan ’s work 😉